pkg/cli/zip.go

```diff
@@ -377,6 +381,71 @@ func maybeAddProfileSuffix(name string) string {
 	return name
 }
 
+type jobTrace struct {
+	jobID   jobspb.JobID
+	traceID int64
+}
+
+// dumpTraceableJobTraces collects the traces for some "traceable" jobs that are
+// in a running state. The job types in this list are the ones that have
+// explicitly implemented the TraceableJob interface.
+func (zc *debugZipContext) dumpTraceableJobTraces() error {
+	ctx := context.Background()
+	rows, err := zc.firstNodeSQLConn.Query(ctx,
+		`SELECT job_id, trace_id FROM crdb_internal.jobs WHERE job_type IN ($1, $2, $3, $4) AND status IN ($5, $6)`,
+		jobspb.TypeBackup.String(),
+		jobspb.TypeRestore.String(),
+		jobspb.TypeImport.String(),
+		jobspb.TypeReplicationStreamIngestion.String(),
+		"running",
+		"reverting",
+	)
+	if err != nil {
+		return err
+	}
+	vals := make([]driver.Value, 2)
+	jobTraces := make([]jobTrace, 0)
+	for err = rows.Next(vals); err == nil; err = rows.Next(vals) {
+		jobID, ok := vals[0].(int64)
+		if !ok {
+			return errors.New("failed to parse jobID")
+		}
+		traceID, ok := vals[1].(int64)
+		if !ok {
+			return errors.New("failed to parse traceID")
+		}
+		jobTraces = append(jobTraces, jobTrace{jobID: jobspb.JobID(jobID), traceID: traceID})
+	}
+
+	func() {
+		// Debug zip collection sets this to false since results from the query are
+		// all dumped into txt files. In our case we parse the results of the query
+		// with their respective types and pre-process the information before
+		// dumping into a zip file.
+		zc.firstNodeSQLConn.SetAlwaysInferResultTypes(true)
+		defer zc.firstNodeSQLConn.SetAlwaysInferResultTypes(false)
+		for _, jobTrace := range jobTraces {
+			inflightTraceZipper := tracezipper.MakeSQLConnInflightTraceZipper(zc.firstNodeSQLConn.GetDriverConn())
```

Comment:
one thing that is slightly sad: the `zip` process is _already contacting every node in the cluster_ so it could, when it does so, get all of that node's trace data from its registry, and by the time zip has contacted every node, it'd have the complete cross-node content of every trace in the cluster... just not assembled into one coherent trace by ID the way this fan-out RPC will assemble it.

Reply:
this doesn't seem worth messing with right now though to be clear.

Reply:
yeah i agree, we already dump the inflight_tracing spans registry of every node in a text file. Maybe there is some solution where we also dump per-node files filtered by trace_id which would mean we'd need the trace_ids plumbed through the table dumping logic. I dunno seems messy, I also wonder whether _anyone_ uses that raw, per-node inflight trace spans table.